client/core/core.go

```diff
@@ -6557,3 +6559,203 @@ func (c *Core) WalletLogFilePath(assetID uint32) (string, error) {
 
 	return wallet.LogFilePath()
 }
+
+func (c *Core) DeleteArchivedRecords(olderThan *time.Time, matchesFile, ordersFile *os.File) error {
+	c.wg.Add(1)
+	c.wg.Done()
+	var (
+		finished  bool
+		err       error
+		csvWriter *csv.Writer
+	)
+
+	// If provided a file to write the orders csv to, write the header and
+	// defer closing the file.
+	if ordersFile != nil {
+		defer ordersFile.Close()
+		csvWriter = csv.NewWriter(ordersFile)
+		csvWriter.UseCRLF = runtime.GOOS == "windows"
+		err = csvWriter.Write([]string{
+			"Host",
+			"Order ID",
+			"Base",
+			"Quote",
+			"Base Quantity",
+			"Order Rate",
+			"Actual Rate",
+			"Base Fees",
+			"Quote Fees",
+			"Type",
+			"Side",
+			"Time in Force",
+			"Status",
+			"Filled (%)",
+			"Settled (%)",
+			"Time",
+		})
+		if err != nil {
+			return fmt.Errorf("error writing orders CSV: %v", err)
+		}
+		csvWriter.Flush()
+		err = csvWriter.Error()
+		if err != nil {
+			return fmt.Errorf("error writing orders CSV: %v", err)
+		}
+	}
+
+	// Delete orders while saving to csv if available until the database
+	// says that's all or context is canceled.
+	for !finished {
+		if err := c.ctx.Err(); err != nil {
+			return fmt.Errorf("unable to delete orders: %v", err)
+		}
+		var ords []*db.MetaOrder
+		ords, finished, err = c.db.DeleteInactiveOrders(olderThan)
+		if err != nil {
+			return fmt.Errorf("unable to delete orders: %v", err)
+		}
+
+		if ordersFile != nil {
+			for _, ord := range ords {
+				cord := coreOrderFromTrade(ord.Order, ord.MetaData)
```

Comment:
Should this ignore cancel orders? I'm thinking it should because it is not very useful information. With these changes they look like
```
,77d91eda537bd722a64fa4d9a45c518a4dc77e3034518b3a5541a39a75c8239e,,,0,0,0,0,0,unknown,buy,immediate,unknown,NaN,NaN,1970-01-01T09:00:00+09:00
```
Which is not very helpful.

Reply:
Since the targeted order would have status cancelled, I tend to agree there's little value in recording the cancel order too.  Only thing I suppose we'd lose is the time of cancellation, right?

BTW, why in that example entry is the timestamp the epoch 0 value?

Reply:
I beleive `coreOrderFromTrade` attempts to return nil for everything if it's a cancel order.

Reply:
Have not changed this yet.

Reply:
I guess I will change `coreOrderFromTrade` to add more data for cancels. Will just tag on another commit...

Reply:
Cancels now look like
```
Host,Order ID,Base,Quote,Base Quantity,Order Rate,Actual Rate,Base Fees,Quote Fees,Type,Side,Time in Force,Status,TargetOrderID,Filled (%),Settled (%),Time
127.0.0.1:17273,54d39189eb33224c5caf50ac759468b240a5a2d247bf433751f3ca87f069cf1a,dcr,eth,10,0.90865,0,0,0,limit,buy,standing,canceled,,0.0,0.0,2022-02-23T16:09:20.916+09:00
127.0.0.1:17273,26916d0a8244c1962b3f823e64c4db52f877de050da26fc1a702999bd535aa63,dcr,eth,0,0,0,0,0,cancel,buy,immediate,no match,54d39189eb33224c5caf50ac759468b240a5a2d247bf433751f3ca87f069cf1a,NaN,NaN,2022-02-23T16:09:31.881+09:00
127.0.0.1:17273,6acf10ae8c7c00411bed447fb51537a911a4cc61999b229a156170ed2d796bab,dcr,eth,10,0.90865,0,0,0,limit,sell,standing,canceled,,0.0,0.0,2022-02-23T16:09:06.299+09:00
127.0.0.1:17273,fe8f94a6c03bc6bf304f023e680f286a2a6dc6f138d0c3ff25d0f4cb3924d6d2,dcr,eth,0,0,0,0,0,cancel,buy,immediate,no match,6acf10ae8c7c00411bed447fb51537a911a4cc61999b229a156170ed2d796bab,NaN,NaN,2022-02-23T16:09:08.914+09:00
```

Added to the third commit.

Reply:
I see the side's default is `buy` so wrong for cancels.

Reply:
Now returning no string for cancel side. Also not attempting to parse filled and settled.

```
127.0.0.1:17273,4d4d35bd86e6c65f7616201b3274b357fcd048ec2c0ce9e9323fa65efc83c387,dcr,eth,0,0,0,0,0,cancel,,immediate,executed,ff99c97afe675dac465dda62b12d93af1198551fd0b8196c4c8ce950ed1bc267,,,2022-02-23T16:31:39.923+09:00
```